cpp/src/io/parquet/page_delta_decode.cu

```diff
@@ -579,17 +580,14 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size)
     __syncthreads();
   }
 
-  // Now turn the array of lengths into offsets, but skip if this is a large string column. In the
-  // latter case, offsets will be computed during string column creation.
+  // Convert string sizes to offsets if this is not a large string column.
   if (not s->col.is_large_string_col) {
-    int value_count = nesting_info_base[leaf_level_index].value_count;
-
-    // if no repetition we haven't calculated start/end bounds and instead just skipped
-    // values until we reach first_row. account for that here.
-    if (!has_repetition) { value_count -= s->first_row; }
-
-    auto const offptr = reinterpret_cast<size_type*>(nesting_info_base[leaf_level_index].data_out);
-    block_excl_sum<decode_block_size>(offptr, value_count, s->page.str_offset);
+    convert_small_string_lengths_to_offsets<decode_block_size>(s, has_repetition);
+  }  // For large strings, update the initial string buffer offset to be used during large string
+     // column construction
+  else {
```

Comment:
I don't have an objective reason for this, I just don't like having the comment in the gap between blocks :D
```suggestion
  } else {
    // For large strings, update the initial string buffer offset to be used during large string
    // column construction
```